app/web/src/components/ModelingView/ConnectionMenuCandidateListItem.vue

```diff
@@ -123,6 +126,24 @@ const props = defineProps({
   controlScheme: { type: String, default: "arrows" },
 });
 
+const typeText = computed(() => {
+  if (candidateIsProp(props.entry)) return "Prop";
+
+  if (!candidateIsSocket(props.entry)) return "?";
```

Comment:
Just confirming: this is impossible, right? Maybe this could be a `candidateType()` method that returns "prop" | "socket" and throws if it's neither? Then you could just have a simple switch () statement here